assets/src/components/routeLadder.tsx

```diff
@@ -37,57 +43,123 @@ const Header = ({ route }: { route: Route }) => {
   reverseLadder,
+  toggleCrowding,
 }: {
+  displayCrowdingToggleIcon: boolean
   ladderDirection: LadderDirection
+  ladderCrowdingToggle: LadderCrowdingToggle
   reverseLadder: () => void
-}) => (
-  <div className="m-route-ladder__controls">
-    <button className="m-route-ladder__reverse" onClick={reverseLadder}>
-      {ladderDirection === LadderDirection.OneToZero
-        ? reverseIcon("m-route-ladder__reverse-icon")
-        : reverseIconReversed("m-route-ladder__reverse-icon")}
-      Reverse
-    </button>
-  </div>
-)
+  toggleCrowding: () => void
+}) => {
+  return (
+    <div className="m-route-ladder__controls">
+      <button className="m-route-ladder__reverse" onClick={reverseLadder}>
+        {ladderDirection === LadderDirection.OneToZero
+          ? reverseIcon("m-route-ladder__reverse-icon")
+          : reverseIconReversed("m-route-ladder__reverse-icon")}
+        Reverse
+      </button>
+      {displayCrowdingToggleIcon &&
+        (ladderCrowdingToggle ? (
+          <button
+            className="m-route-ladder__crowding-toggle m-route-ladder__crowding-toggle--hide"
+            onClick={toggleCrowding}
+          >
+            {crowdingIcon(
+              "m-route-ladder__crowding-toggle-icon m-route-ladder__crowding-toggle-icon"
+            )}
+            Hide riders
+          </button>
+        ) : (
+          <button
+            className="m-route-ladder__crowding-toggle m-route-ladder__crowding-toggle--show"
+            onClick={toggleCrowding}
+          >
+            {crowdingIcon(
+              "m-route-ladder__crowding-toggle-icon m-route-ladder__crowding-toggle-icon"
+            )}
+            Show riders
+          </button>
+        ))}
+    </div>
+  )
+}
+
+const someVehicleHasCrowding = (
+  vehiclesAndGhosts?: VehicleOrGhost[]
+): boolean => {
+  if (vehiclesAndGhosts === undefined) {
+    return false
+  }
+
+  const vehicleWithCrowding = vehiclesAndGhosts.find(
+    (vehicleOrGhost) =>
+      isVehicle(vehicleOrGhost) &&
+      vehicleOrGhost.hasOwnProperty("crowding") &&
+      vehicleOrGhost.crowding !== null
+  )
+
+  return !!vehicleWithCrowding
+}
 
 const RouteLadder = ({
   route,
   timepoints,
   vehiclesAndGhosts,
   selectedVehicleId,
+  crowdingEnabled,
 }: Props) => {
-  const [{ ladderDirections }, dispatch] = useContext(StateDispatchContext)
+  const [{ ladderDirections, ladderCrowdingToggles }, dispatch] = useContext(
+    StateDispatchContext
+  )
   const ladderDirection = getLadderDirectionForRoute(ladderDirections, route.id)
   const reverseLadder = () => {
     dispatch(flipLadder(route.id))
   }
 
+  const ladderCrowdingToggle = getLadderCrowdingToggleForRoute(
+    ladderCrowdingToggles,
+    route.id
+  )
+  const toggleCrowding = () => {
+    dispatch(toggleLadderCrowding(route.id))
+  }
+
   const byPosition: VehiclesByPosition = groupByPosition(
     vehiclesAndGhosts,
     route.id,
     ladderDirection
   )
 
+  const displayCrowding = !!(
```

Comment:
Oh, more importantly! The crowding data is shown whether it's toggled on for the route or not. You need to check the state of the toggle here, too.

<img width="375" alt="Screen Shot 2020-07-13 at 15 37 18" src="https://user-images.githubusercontent.com/23065557/87346054-05c9f100-c51f-11ea-84e6-860ede65cdf0.png">

Reply:
Fixed the toggle. As for the `!!` note that `crowdingEnabled` can be `undefined` as well. We probably don't really need to do the cast at all, though...

Reply:
@skyqrose I remember why I did it that way. I wanted to simplify the implementation of `Controls`, etc. a bit by having them take strictly booleans, rather than `boolean | undefined`. So I'm inclined to keep it as-is.